amp/amp-service-worker-offline-commenting.js

```diff
@@ -0,0 +1,51 @@
+/* global ERROR_MESSAGES, SITE_URL */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'amp-wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+		const clone = event.request.clone();
+		return fetch( event.request )
+			.then( ( response ) => {
+				// @todo Make sure that 409 etc. error still work as expected.
+				return response;
+			} )
+			.catch( () => {
+				const bodyPromise = clone.blob();
+				return bodyPromise.then(
+					function( body ) {
+						const request = event.request;
+						const req = new Request( request.url, {
+							method: request.method,
+							headers: request.headers,
+							mode: 'same-origin',
```

Comment:
I think that `mode` will need to be set to `cors` since when submitting a comment from the AMP Cache it will need to connect to the origin.